packages/cashier/src/pages/payment-agent/payment-agent-deposit-withdraw-container/payment-agent-deposit-withdraw-container.jsx

```diff
@@ -5,40 +5,90 @@ import { DesktopWrapper, Dropdown, MobileWrapper, SelectNative, Text } from '@de
 import { localize, Localize } from '@deriv/translations';
 import { connect } from 'Stores/connect';
 import PaymentAgentCard from '../payment-agent-card';
+import PaymentAgentWithdrawConfirm from '../payment-agent-withdraw-confirm';
+import PaymentAgentUnlistedWithdrawForm from '../payment-agent-unlisted-withdraw-form';
+import PaymentAgentReceipt from '../payment-agent-receipt';
 import PaymentAgentDisclaimer from '../payment-agent-disclaimer';
+import SideNote from 'Components/side-note';
 
 const PaymentAgentDepositWithdrawContainer = ({
+    app_contents_scroll_ref,
     is_dark_mode_on,
     is_deposit,
+    is_try_withdraw_successful,
+    is_withdraw_successful,
     onChangePaymentMethod,
     payment_agent_list,
+    resetPaymentAgent,
     selected_bank,
     supported_banks,
+    verification_code,
 }) => {
-    const list_with_default = [
-        { text: <Localize i18n_default_text='All payment methods' />, value: 0 },
-        ...supported_banks,
-    ];
+    React.useEffect(() => {
+        return () => {
+            if (!is_deposit) {
+                resetPaymentAgent();
+            }
+        };
+    }, [resetPaymentAgent]);
 
     React.useEffect(() => {
         return () => {
             onChangePaymentMethod({ target: { value: '0' } });
         };
-    }, [onChangePaymentMethod]);
+    }, []);
```

Comment:
```suggestion
    }, [onChangePaymentMethod]);
```
@george-usynin-binary Most of the `useEffect` hooks have missing dependencies, Shouldn’t we fix them? 🤔

Reply:
@farzin-fs not sure why we need to add this onChangePaymentMethod into dependency. the main idea is to reset dropdown list to default value (All payment methods) when we leave the payment-agent page. and as i realise onChangePaymentMethod always be the same function because when the app is loading, there will be created an instance of the payment-agent store and we will use this instance during the whole lifecycle of the app.

Reply:
in this case i decided to disable this rule for entire file to fix the warning because to be honest I don't see the need to add onChangePaymentMethod to dependencies, again because of `onChangePaymentMethod` immutability